src/pool/srv_pool.c

```diff
@@ -2173,6 +2182,8 @@ ds_pool_connect_handler(crt_rpc_t *rpc)
 
 	hdl.ph_flags = in->pci_flags;
 	hdl.ph_sec_capas = sec_capas;
+	strncpy(hdl.ph_machine, machine, MAXHOSTNAMELEN);
+
```

Comment:
any possibility that machine source string might (maybe only if there were to be a bug elsewhere?) not have a null termination character? Should hdl.ph_machine[MAXHOSTNAMELEN] = '\0'; be added following the strncpy() call?

Reply:
ph_machine is MAXHOSTNAMELEN+1 and the way strncpy works this will put a \0 at the end if one isn't found sooner than MAXHOSTNAMELEN. So it will always be null terminated whether its done here in strncpy or was done earlier when the buffer was initalized to 0s.

Reply:
It seems like from the manpage, with strncpy() it will not add a null byte if one is not present in the source string in the first (in this example) MAXHOSTNAMELEN bytes, so the hdl = {0} initializer I guess covers it in this case.

Reply:
Correct. I've been fixing a lot of issues in the code related to this quirky behavior in strncpy